2.6.md

```diff
@@ -2289,53 +2318,56 @@ This object provides information pertaining to the device through which the user
     <td><code>mccmnc</code></td>
     <td>string</td>
     <td>Mobile carrier as the concatenated MCC-MNC code (e.g., “310-005” identifies Verizon Wireless CDMA in the USA). Refer to https://en.wikipedia.org/wiki/Mobile_country_code for further examples. Note that the dash between the MCC and MNC parts is required to remove parsing ambiguity. The MCC-MNC values represent the SIM installed on the device and do not change when a device is roaming. Roaming may be inferred by a combination of the MCC-MNC, geo, IP and other data signals.</td>
- <tr>
-  </tr>
+   </tr>
+  <tr>
     <td><code>connectiontype</code></td>
     <td>integer</td>
     <td>Network connection type. Refer to <a href="https://github.com/InteractiveAdvertisingBureau/AdCOM/blob/master/AdCOM%20v1.0%20FINAL.md#list--connection-types-">List: Connection Types</a> in AdCOM 1.0.</td>
- <tr>
   </tr>
+  <tr>
     <td><code>ifa</code></td>
     <td>string</td>
     <td>ID sanctioned for advertiser use in the clear (i.e., not hashed).</td>
-  <tr>
   </tr>
+    <tr>
+    <td><code>ifa_type</code></td>
+    <td>integer</td>
+    <td>Declared source of the device.ifa provided in the bid request. Non-device IDs should be placed in eids. Refer to <a href="https://github.com/InteractiveAdvertisingBureau/AdCOM/blob/master/AdCOM%20v1.0%20FINAL.md#list--ifa_types-">List: IFA Types</a> for values.</td>
```

Comment:
Suspect that this is the link you intended for IFA types: https://github.com/InteractiveAdvertisingBureau/AdCOM/blob/develop/AdCOM%20v1.0%20FINAL.md#list_ifa_types (granted, this will eventually get merged into _main_).

More importantly, this list of IFA types proposed (dpid, ppid, sspid, sessionid) falls short of the way that the extension (device.ext.ifa_type) is commonly used today. According to the Tech Lab's [IFA guidelines](https://iabtechlab.com/wp-content/uploads/2018/12/OTT-IFA-guidelines.final_Dec2018.pdf), other acceptable values include rida, aaid, idfa, afai, etc. And in practice, we see additional IFA type values in the bid request stream as new platform IDs begin to make their way through programmatic pipes (e.g. vida, tifa). These surely won't be the last additions to the list either.

So it seems like either we need to:
- maintain an authoritative list of platform IDs that's larger than what's proposed in AdCOM now
- provide some standardized, platform-independent values (like the current list) but also allow for platform-specific values to be used
- guide implementers to use `dpid` for any device-provided ID, but then break out some new field to represent which device or ads framework on that device actually generated the ID (i.e. its namespace)

Would like @jwilhite-ias @emailcurt @iantri and others to comment here.

Reply:
Also @ChrisBasis .

Reply:
I hear from @hillslatt that this proposal explicitly nuked the platform-specific values (i.e. not an oversight). This leads to the question of whether it's possible to have more than 1 IFA namespace per device? For example, could you have multiple apps, each using a different ads framework, each generating a different device-provided ID for their respective ad requests? Or does device-provided ID specifically mean OS-provided?